### PR TITLE
chore(gcp_chronicle): allow enabling google chronicle separately from…

### PR DESCRIPTION
**Pull request by pront**

### CODE REVIEW

**Comment by pront** [comment]
Pending: https://github.com/vectordotdev/vector/actions/runs/9213626518

File: src/sinks/gcp_chronicle/mod.rs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#[cfg(any(feature = "sinks-gcp-chronicle", feature = "sinks-gcp"))]
```
Comment (jszwedko) [inline src/sinks/gcp_chronicle/mod.rs:1]: Could we instead have `sinks-gcp` depend on `sinks-gcp-chronicle`?

Comment (jszwedko) [reply src/sinks/gcp_chronicle/mod.rs:1]: Yep!

File: Cargo.toml [hunk 2]
```diff
@@ -736,7 +736,8 @@ sinks-datadog_metrics = ["protobuf-build", "dep:prost-reflect"]
 sinks-datadog_traces = ["protobuf-build", "dep:rmpv", "dep:rmp-serde", "dep:serde_bytes"]
 sinks-elasticsearch = ["transforms-metric_to_log"]
 sinks-file = ["dep:async-compression"]
-sinks-gcp = ["dep:base64", "gcp"]
+sinks-gcp = ["sinks-gcp-chronicle"]
+sinks-gcp-chronicle =  ["dep:base64", "gcp"]
```
Comment (jszwedko) [inline Cargo.toml:740]:
```suggestion
sinks-gcp = ["sinks-gcp-chronicle", dep:base64", "gcp"]
sinks-gcp-chronicle =  ["gcp"]
```

Nit: I don't think `sinks-gcp-chronicle` actually depends on `base64`, only `gcp_pubsub` does.

**Comment by pront** [comment]
Pending: https://github.com/vectordotdev/vector/actions/runs/9214130471/job/25349627956

File: src/sinks/mod.rs [hunk 3]
```diff
@@ -51,7 +51,9 @@ pub mod elasticsearch;
 pub mod file;
 #[cfg(feature = "sinks-gcp")]
 pub mod gcp;
-#[cfg(feature = "sinks-gcp")]
+#[cfg(feature = "sinks-gcp-chronicle")]
+pub mod gcp_chronicle;
+#[cfg(any(feature = "sinks-gcp-chronicle", feature = "sinks-gcp"))]
```
Comment (jszwedko) [inline src/sinks/mod.rs:56]:
```suggestion
#[cfg(feature = "sinks-gcp")]
```

Does chronicle depend on gcs_common? That suprises me.

File: Cargo.toml [hunk 4]
```diff
@@ -736,7 +736,8 @@ sinks-datadog_metrics = ["protobuf-build", "dep:prost-reflect"]
 sinks-datadog_traces = ["protobuf-build", "dep:rmpv", "dep:rmp-serde", "dep:serde_bytes"]
 sinks-elasticsearch = ["transforms-metric_to_log"]
 sinks-file = ["dep:async-compression"]
-sinks-gcp = ["dep:base64", "gcp"]
+sinks-gcp = ["sinks-gcp-chronicle", dep:base64", "gcp"]
```
Comment (jszwedko) [inline Cargo.toml:739]:
```suggestion
sinks-gcp = ["sinks-gcp-chronicle", "dep:base64", "gcp"]
```

I had a typo 😓

File: src/sinks/mod.rs (hunk 3 above)
Comment (pront) [reply src/sinks/mod.rs:56]: Yes it looks like it: https://github.com/vectordotdev/vector/blob/master/src/sinks/gcp/chronicle_unstructured.rs#L35

File: Cargo.toml (hunk 2 above)
Comment (pront) [reply Cargo.toml]: Correct, committed this suggestion (and the typo fix).

File: src/sinks/mod.rs (hunk 3 above)
Comment (jszwedko) [reply src/sinks/mod.rs:56]: Ah, it seems like the reference types there should probably be in `gcp_common` but that's definitely out-of-scope of this PR 👍

**Review by jszwedko — APPROVED** [review]

**Comment by pront** [comment]
Pending: https://github.com/vectordotdev/vector/actions/runs/9214154297

**Review by jszwedko — APPROVED** [review]
Assuming CI passes 👍

**Pull request merged**